pkg/js/helpers.js

```diff
@@ -312,6 +321,33 @@ function addRecord(d,type,name,target,mods) {
     return rec;
 }
 
+function addRecordSRV(d,type,name,srvpriority,srvweight,srvport,target,mods) {
+    var rec = {type: type, name: name, srvpriority: srvpriority, srvweight: srvweight, srvport: srvport, target: target, ttl:d.defaultTTL, meta:{}};
+    // for each modifier, decide based on type:
+    // - Function: call is with the record as the argument
+    // - Object: merge it into the metadata
+    // FIXME(tlim): Factor this code out to its own function.
```

Comment:
I'm not confident in my javascript abilities to do that.  I'll file an issue and we'll get to this later.